backends/ebpf/README.md

```diff
@@ -335,4 +335,75 @@ The following tests run ebpf programs:
    Note: by default the kernel ebpf tests are disabled; if you want to enable them
    you can modify the file `backends/ebpf/CMakeLists.txt` by setting this variable to `True`:
    `set (SUPPORTS_KERNEL True)`
-   
\ No newline at end of file
+   
+# How to inject custom extern function to the generated eBPF program?
+
+The P4 to eBPF compiler comes with the support for custom C extern functions. It means that a developer
+can write a custom, eBPF-compatible (acceptable by BPF verifier) C function and call it
+from the P4 program as a normal P4 action. As a result, P4 program can be extended with
+functionality, which is not supported natively by the P4 language. This feature is briefly
+described below.
+
+## Basic principles 
+
+The C extern function can effectively enhance the functionality of P4 program. A programmer should be able to write own function, declare it in the P4 program and invoke from within P4 action or P4 control block. 
+
+A definition of the C extern function should follow the strict rules, which we call a calling convention. The calling convention describes how basic data types are passed between P4 and C, how data types are converted, how direction of P4 extern’s parameter is handled in C, etc.
```

Comment:
This paragraph is probably unnecessary; the section below covers already all of this.

Reply:
Done.